sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -320,11 +320,13 @@ Command *Scheduler::GraphBuilder::addHostAccessor(Requirement *Req) {
 
   AllocaCommandBase *SrcAllocaCmd = nullptr;
 
-  if (Record->MAllocaCommands.empty())
+  if (Record->MAllocaCommands.empty() || IsSuitableSubReq(Req))
```

Comment:
Another approach here might be to check if SrcAllocaCmd is null and create it the same way getOrCreateAllocaForReq does if IsSuitableSubReq(Req) returns true. Thing is, this approach would violate code reuse.

Reply:
The second commit implements this second approach.